### ISSUE TITLE
PSA - default parameters and implicit conversions are evil

### ISSUE THREAD

**Issue by ilteroi** [LoneGazebo/Community-Patch-DLL#8308]
i found a number of bugs caused by implicit type conversions. they were hidden because warnings were disabled ... but others might still be there because for "widening" bool to int conversion visual c++ doesn't even have a warning. 

this is especially insidious in combination with default parameters, because somebody might add an additional default parameter in the middle of a parameter list and it will compile without adapting the callers!

anyway, long story short, don't do this and i broke savegames in 8bf626f53e5cf6c789c45f598aa3c5764e0e27e6.

**Comment by LoneGazebo** [comment]
where did you alter the code? I don't see a commit on this topic.
